### DISCUSSION TITLE
Provide CloudWatch-compatible interval variable

### DISCUSSION THREAD

**Discussion by rehevkor5** [grafana/grafana#44489] [Dashboards, Panels & Visualizations]
### What would you like to be added:

As suggested in https://github.com/grafana/grafana/issues/20262 and https://github.com/grafana/grafana/issues/36447 Grafana should provide a version of `$__interval` which is compatible with CloudWatch so that it can be used in the `Period` parameter of `SEARCH()` expressions. Search expression syntax: https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/search-expression-syntax.html

### Why is this needed (describe your use case and goals):

My use case (though there are likely others), is in the visualization of CloudWatch metrics which use `Count` as their units. These kind of metrics, if visualized simply as a `Sum`, provide misleading/confusing results because the values vary based on the length of the `$__interval`. If zoomed in, you might have an interval of 1m and so the value of the metric will be low. When you zoom out, the value will get larger because it's summing over a longer time period. This makes it impossible for the graph to use a constant scale and confuses people.

It is much better to instead follow the advice in the CloudWatch documentation. For example, see "ConsumedReadCapacityUnits" on https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/metrics-dimensions.html

> Use the Sum statistic to calculate the consumed throughput. For example, get the Sum value over a span of one minute, and divide it by the number of seconds in a minute (60) to calculate the average ConsumedReadCapacityUnits per second (recognizing that this average does not highlight any large but brief spikes in read activity that occurred during that minute). You can compare the calculated value to the provisioned throughput value that you provide DynamoDB.

It is possible to hard-code the interval, such as 60, in the query, or use an ad-hoc Custom Variable such as `1m : 60, 5m : 300, 10m : 600`. But then you do not benefit from Grafana's automatic interval calculation, which is problematic. And, there may be unexpected interactions between the query's interval vs. grafana's interval. For example, if you specify an interval of 60, but Grafana actually queries with a longer Period such as 600, I think the results would be incorrect due to dividing by the wrong number.

It is not possible to use `$__interval` or `$__interval_ms` directly because they are not in seconds as expected by `SEARCH`. Also, you cannot perform a calculation such as `${__interval_ms}/1000` because the query will be rejected:

Example expression sent to the server: `"SEARCH(' {AWS/DynamoDB,TableName} MetricName=\"ConsumedReadCapacityUnits\" TableName=mytable ', 'Sum', 60000/1000)/60000/1000"`

Response:

> ValidationError: SEARCH arguments must be '[String, String, Positive Number]' status code: 400, request id: 49841285-161f-4b41-be4b-fc526382f6b3

 It's worth noting that https://grafana.com/docs/grafana/latest/datasources/aws-cloudwatch/ mentions the existence of `{{period}}`, which appears to be what we want. But, it cannot be used inside the Expression. Also, at least in my version of Grafana (v7.3.5) it can take on unusable values such as `900)/900`.

**Comment by justinTM** [comment]
Along with a variety of other issues Grafana's team has designated WONTFIX, CloudWatch in Grafana across the board has been buggy, poorly-documented, and missing features.

**Reply by onitake** [reply]
I don't think that this is a valid justification for not supporting a specific requested feature.
And I also don't think it's a correct statement either, because the CloudWatch module works all right in general.
Why would perceived brokenness prevent improvement?

**Comment by rehevkor5** [comment]
Per https://github.com/grafana/grafana/discussions/41811#discussioncomment-7525315 the docs now mention `$__period_auto` which appears to probably be what I was asking for. I don't use Grafana any more, but I will assume that works and close this as "outdated".
